pkg/cmd/pr/merge/merge.go

```diff
@@ -370,11 +370,11 @@ func (m *mergeContext) merge() error {
 
 // Delete local branch if requested and if allowed.
 func (m *mergeContext) deleteLocalBranch() error {
-	if m.crossRepoPR || m.autoMerge {
+	if m.autoMerge {
 		return nil
 	}
 
-	if m.merged {
+	if !m.crossRepoPR && m.merged {
```

Comment:
What is the reason for checking if it is a cross repo PR? Seems like that is no longer relevant.

Reply:
Not have this check made the test `TestPrMerge_alreadyMerged_withMergeStrategy_crossRepo` crashing but I was able to update the test correctly, removing the need for a check.